src/sagemaker/serve/model_server/tensorflow_serving/server.py

```diff
@@ -0,0 +1,139 @@
+"""Module for Local Tensorflow Server"""
+
+from __future__ import absolute_import
+
+import requests
+import logging
+import platform
+from pathlib import Path
+from sagemaker.base_predictor import PredictorBase
+from sagemaker.session import Session
+from sagemaker.serve.utils.exceptions import LocalModelInvocationException
+from sagemaker.s3_utils import determine_bucket_and_prefix, parse_s3_url
+from sagemaker import fw_utils
+from sagemaker.serve.utils.uploader import upload
+from sagemaker.local.utils import get_docker_host
+
+logger = logging.getLogger(__name__)
+
+
+class LocalTensorflowServing:
+    """LocalTensorflowServing class."""
+
+    def _start_tensorflow_serving(
+        self, client: object, image: str, model_path: str, secret_key: str, env_vars: dict
+    ):
+        """Starts a local tensorflow serving container.
+
+        Args:
+            client: Docker client
+            image: Image to use
+            model_path: Path to the model
+            secret_key: Secret key to use for authentication
+            env_vars: Environment variables to set
+        """
+        self.container = client.containers.run(
+            image,
+            "serve",
+            detach=True,
+            auto_remove=True,
+            network_mode="host",
+            volumes={
+                Path(model_path): {
+                    "bind": "/opt/ml/model",
+                    "mode": "rw",
+                },
+            },
+            environment={
+                "SAGEMAKER_SUBMIT_DIRECTORY": "/opt/ml/model/code",
+                "SAGEMAKER_PROGRAM": "inference.py",
+                "SAGEMAKER_SERVE_SECRET_KEY": secret_key,
+                "LOCAL_PYTHON": platform.python_version(),
+                **env_vars,
```

Comment:
nit: let's ensure no duplicate env variables.

Reply:
Curious, why do we need the extra env_vars here?

Reply:
I was following the same behavior as in torchserve inference script. I think the rationale is that we want to allow customers to alternate inference script if they want to and extra env vars can be picked up by their own inference script.